MSSQL/OMOPLoader.sql

```diff
@@ -1613,43 +1613,93 @@ IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'[dbo].[build
+-- Input parameters: @domain -- domain the temp table modifier is used for (must match value in i2o_config_modifier.c_domain
+--                   @force_update -- Forces rebuilding of temp tables
+-- Notes: Temp tables persist only during a connection
+--        This procedure defaults to NOT overwriting a pre-existing temp table
+--        If you need the temp tables rebuilt during a single connection set parameter @force_update = 1
+---------------------------------------------------------------------------------
+--Variables
 declare @sqltext nvarchar (2000)
 declare @tablename varchar(50)
 declare @path varchar(400)
 declare @target_column varchar(50)
+declare @temp_tablename varchar(100)
+declare @temp_tableidx varchar(100)
 
+--Cursors
 declare getconfigsql cursor local for
 select c_tablename, c_path, c_target_column from i2o_config_modifier where c_domain=@domain
 
 begin
+	--Loop Cursor
     open getconfigsql;
     fetch next from getconfigsql into @tablename,@path,@target_column;
     while @@fetch_status=0
     begin
-        set @sqltext = 'select patient_num, encounter_num, m.provider_id, concept_cd, start_date, o.i_stdcode  ' +@target_column+' ' +
-            'into temp_mod_'+@target_column+' from i2b2fact M ' +
-            --inner join visit_occurrence enc on enc.person_id = m.patient_num and enc.visit_occurrence_id = m.encounter_Num
-            ' inner join '+@tablename+' o on m.modifier_cd =o.c_basecode ' +
-            ' where c_fullname LIKE '''+@path+ '%'''
-        exec sp_executesql @sqltext
-        raiserror('Modifiers: %s',0,1,@sqltext) with NOWAIT;
+
+		--Set table temp name and idx name
+		set @temp_tablename = 'temp_mod_' + @target_column;
+		set @temp_tableidx = @temp_tablename + '_cidx';
+
+		-- If force_update then drop existing temp table if it exists
+		IF @force_update = 1 
+			BEGIN
+					IF OBJECT_ID(@temp_tablename) IS NOT NULL
+						BEGIN TRY
+							set @sqltext = 'DROP TABLE ' + @temp_tablename;
+							exec sp_executesql @sqltext;
+						END TRY
+						BEGIN CATCH
+							RAISERROR('ERROR: Failed to execute %s', 0, 1, @sqltext) with NOWAIT;
+						END CATCH;
+			END;
+
+		-- If the temp table does not already exist create it and index it
+		IF OBJECT_ID(@temp_tablename) IS NULL
+			BEGIN TRY
+				set @sqltext = 'select patient_num, encounter_num, m.provider_id, concept_cd, start_date, instance_num, o.i_stdcode  ' +@target_column+' ' +
+					'into temp_mod_'+@target_column+' from i2b2fact M ' +
+					' inner join '+@tablename+' o on m.modifier_cd =o.c_basecode ' +
+					' where c_fullname LIKE '''+@path+ '%'''
+				exec sp_executesql @sqltext
+				set @sqltext = 'create clustered index ' + @temp_tableidx + ' on ' + @temp_tablename + '(patient_num, encounter_num, provider_id, concept_cd, start_date, instance_num)'
+				exec sp_executesql @sqltext
+				PRINT 'CREATED TEMPORARY TABLE ' + @TEMP_TABLENAME;

```

Comment:
FYI PRINT often does print out until the very end...

Reply:
Ok, think that's fine for now. Maybe I'll change it to a RAISERROR later